addons/mail/models/mail_tracking_duration_mixin.py

```diff
@@ -110,3 +118,114 @@ def _get_duration_from_tracking(self, trackings):
+      - _resource_is_not_rotting_hook(task) must be overriden to add additional conditions for which a stage IS NOT rotting
+        (e.g. a task that has been closed):
+            def _resource_is_not_rotting_hook(self, task):
+                if task.is_closed:
+                    return True
+                return super()._resource_is_not_rotting_hook(task)
+      - _search_is_rotting() must be overriden to update the returned domain with additional conditions for which a stage COULD BE rotting
+        (and condition, all need to be true):
+            def _search_is_rotting(self, operator, value):
+                sup = super()._search_is_rotting(operator, value)
+                return Domain.AND([sup, [('is_closed', '=', True)]])
+
+    - The date_rot, is_rotting, day_rotting, last_activity fields need to be added to the relevant views
+        (as well as the "days to rot" field on the tracking model).
+        You may want to use the rotting_form and rotting_kanban field widgets to display the fields visually on form and kanban view
+        (please note- these widgets need both day_rotting and is_rotting fields on the view to function).
+
+
+    Note that if _stage_day_rot_field is not set, or if the value stored by the field pointed by _stage_day_rot_field is 0,
+    then the resource will never rot.
+    """
+
+    date_rot = fields.Date('Date on which this resource will start rotting', compute="_compute_date_rot", store=True)
+    is_rotting = fields.Boolean('Rotting', compute='_compute_rotting', search='_search_is_rotting')
+    day_rotting = fields.Integer('Days Rotting', help='Day count since this resource was last updated',
+        compute='_compute_rotting')
+    last_activity = fields.Date('Date of last activity', compute="_compute_last_activity", store=True, readonly=False)
+
+    @api.depends('write_date')
+    def _compute_last_activity(self):
+        for resource in self:
+            resource.last_activity = self.env.cr.now()
+
+    def _get_day_rot_value(self):
+        self.ensure_one()
+        rotting_stage = self[self._track_duration_field]
+        if not rotting_stage or not hasattr(self, '_stage_day_rot_field'):
+            # If _stage_day_rot_field has not been set, the rotting feature is not enabled for this model
+            return 0
+        if not self._stage_day_rot_field in rotting_stage:
+            raise UserError('Models using the rotting feature need to declare a "day_rot" field on their stage model')
+        day_rot = rotting_stage[self._stage_day_rot_field]
+        return day_rot
+
+    @api.depends('last_activity', 'write_date')
+    def _compute_date_rot(self):
+        for resource in self:
+            day_rot = resource._get_day_rot_value()
+            if not day_rot:
+                continue
+            last_activity = resource.last_activity or resource.write_date or self.env.cr.now()
+            resource.date_rot = last_activity + timedelta(day_rot)
```

Comment:
`timedelta(days=day_rot)`, generally speaking we try to avoid using positional arguments to fill in named ones.